python/tests/unit/prompt_template/test_prompt_template_e2e.py

```diff
@@ -127,27 +139,59 @@ async def test_it_allows_to_pass_escaped_values2_to_functions(self, kernel: Kern
 
         # Act
         result = await KernelPromptTemplate(
-            prompt_template_config=PromptTemplateConfig(name="test", description="test", template=template)
+            prompt_template_config=PromptTemplateConfig(
+                name="test", description="test", template=template, allow_dangerously_set_content=True
+            )
         ).render(kernel, None)
 
         # Assert
         assert '== a"b != 123 ==' == result
 
     @mark.asyncio
-    @mark.parametrize("template,expected_result", [(t, r) for t, r in _get_template_language_tests()])
-    async def test_it_handle_edge_cases(self, kernel: Kernel, template: str, expected_result: str):
+    @mark.parametrize("template,expected_result", [(t, r) for t, r in _get_template_language_tests(safe=False)])
+    async def test_it_handle_edge_cases_unsafe(self, kernel: Kernel, template: str, expected_result: str):
+        # Arrange
+        kernel.add_plugin(MyPlugin(), "my_plugin")
+
+        # Act
+        if expected_result.startswith("ERROR"):
+            with raises(TemplateSyntaxError):
+                await KernelPromptTemplate(
+                    prompt_template_config=PromptTemplateConfig(name="test", description="test", template=template),
+                    allow_dangerously_set_content=True,
+                ).render(kernel, KernelArguments())
+        else:
+            result = await KernelPromptTemplate(
+                prompt_template_config=PromptTemplateConfig(name="test", description="test", template=template),
+                allow_dangerously_set_content=True,
+            ).render(kernel, KernelArguments())
+
+            # Assert
+            assert expected_result == result
+
+    @mark.asyncio
+    @mark.parametrize("template,expected_result", [(t, r) for t, r in _get_template_language_tests(safe=True)])
+    async def test_it_handle_edge_cases_safe(self, kernel: Kernel, template: str, expected_result: str):
         # Arrange
         kernel.add_plugin(MyPlugin(), "my_plugin")
 
         # Act
         if expected_result.startswith("ERROR"):
             with raises(TemplateSyntaxError):
                 await KernelPromptTemplate(
-                    prompt_template_config=PromptTemplateConfig(name="test", description="test", template=template)
+                    prompt_template_config=PromptTemplateConfig(
+                        name="test",
+                        description="test",
+                        template=template,
+                    )
                 ).render(kernel, KernelArguments())
         else:
             result = await KernelPromptTemplate(
-                prompt_template_config=PromptTemplateConfig(name="test", description="test", template=template)
+                prompt_template_config=PromptTemplateConfig(
+                    name="test",
+                    description="test",
+                    template=template,
+                )
```

Comment:
I'm looking for the new tests corresponding to these ones: https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/SemanticKernel.UnitTests/PromptTemplate/KernelPromptTemplateTests.cs#L522-L951

Reply:
added now to test_prompt_template_e2e, I did notice we have some inconsistency in what is tested where...